executor/builder.go

```diff
@@ -2655,7 +2707,23 @@ func (b *executorBuilder) buildIndexLookUpReader(v *plannercore.PhysicalIndexLoo
 	sctx := b.ctx.GetSessionVars().StmtCtx
 	sctx.IndexNames = append(sctx.IndexNames, is.Table.Name.O+":"+is.Index.Name.O)
 	sctx.TableIDs = append(sctx.TableIDs, ts.Table.ID)
-	return ret
+
+	// TODO: Remove the following 3 lines code when the code is full implemented.
+	if !tryNewPartitionImplementation(b.ctx) {
+		return ret
+	}
+
+	if pi := is.Table.GetPartitionInfo(); pi == nil {
+		return ret
+	}
+
+	nextPartition := nextPartitionForIndexLookUp{exec: ret}
+	exec, err := buildPartitionTable(b, ts.Table, v.PruningConds, ret, nextPartition)
```

Comment:
maybe have same question,  when change test `where p.id = t.id` to `where p.c = t.id`

Reply:
It's a bit more difficult because the the reader reads chunk by chunk. Inside a chunk, each row may correspond to diffierent table. 
The current TableReader/IndexReader/IndexLookUp  have an assumption that the underlying table is fixed and not change, so the build key range logic need to change in those three reader executor.

I plan to improve here later, not in this PR.
Although the partition pruning is not done for some case, the code should be able to run.